Documentation/gettingstarted/encryption-wireguard.rst

```diff
@@ -0,0 +1,111 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _encryption_wg:
+
+********************************
+Wireguard Transparent Encryption
+********************************
+
+This guide explains how to configure Cilium with transparent encryption of
+traffic between Cilium-managed endpoints using `Wireguard <https://www.wireguard.com/>`_.
+
+When Wireguard is enabled in Cilium, the agent running on each cluster node
+will establish a secure Wireguard tunnel between it and all other known nodes
+in the cluster. Each node automatically creates its own encryption key-pair and
+distributes its public key via the ``io.cilium.network.wg-pub-key`` annotation in
+the Kubernetes ``CiliumNode`` custom resource object. Each node's public key is then
+used by other nodes to decrypt and encrypt traffic from and to Cilium-managed
+endpoints running on that node.
+
+The Wireguard tunnel endpoint is exposed on UDP port ``51871`` on each node. If
+you run Cilium in an environment that requires firewall rules to enable
+connectivity, you will have to ensure that all Cilium cluster nodes can reach
+each other via that port.
+
+Enable Wireguard in Cilium
+==========================
+
+Before you enable Wireguard in Cilium, please ensure that the Linux distribution
+running on your cluster nodes has support for Wireguard in kernel mode
+(i.e. ``CONFIG_WIREGUARD=m`` on Linux 5.6 and newer, or via the out-of-tree
+Wireguard module on older kernels).
+
+.. tabs::
+
+    .. group-tab:: Cilium CLI
+
+       If you are deploying Cilium with the Cilium CLI, pass the following
+       options:
+
+       .. code-block:: shell-session
+
+          cilium install --config enable-wireguard=true --config enable-l7-proxy=false
+
+    .. group-tab:: Helm
+
+       If you are deploying Cilium with Helm by following
+       :ref:`k8s_install_helm`, pass the following options:
+
+       .. parsed-literal::
+
+           helm install cilium |CHART_RELEASE| \\
+             --namespace kube-system \\
+             --set l7Proxy=false \\
+             --set wireguard.enabled=true
```

Comment:
Reviewer note: This is functional as is, but might changed in https://github.com/cilium/cilium/pull/15628